### DISCUSSION TITLE
openssl command to sign and verify a zip file and then extraxt the original zip file from the signed zip file

### DISCUSSION THREAD

**Discussion by Aravindms369** [openssl/openssl#21664] [Q&A]
I have tried "smime" method for signing and verification of zip file:

signing
-----------
openssl smime -sign -in payload_new.zip -out your_signed_zip_file.zip -signer client_cert.pem -inkey client_private_key.pem -certfile ca_cert.pem

verification
-----------------
openssl smime -verify -in your_signed_zip_file.zip -out verified_zip_file.zip -CAfile ca_cert.pem

output: Verification successful

after verification when i try to unzip the verified zip file to get the original payload contents i amgetting below error:
> unzip verified_zip_file.zip

**Archive:  verified_zip_file.zip
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of verified_zip_file.zip or
        verified_zip_file.zip.zip, and cannot find verified_zip_file.zip.ZIP, period.**

Please help me to resolve this issue

**Comment by slontis** [comment]
I would suggest looking at the options supplied by doing
````
openssl smime -help
````
Should inform be DER?
and do you need the binary option?

You could also look at the SHA checksum of the decrypted file to see if it matches your original file.. i.e
````
openssl sha256  file.zip
````
